packages/framework-provider-azure-infrastructure/src/infrastructure/index.ts

```
@@ -1,39 +1,76 @@
+  logger.info('Zip file uploaded')
+}
+
 /**
  * Deploys the application in Azure
  */
 async function deployApp(logger: Logger, config: BoosterConfig): Promise<void> {
-  const credentials = await azureCredentials()
-  const [resourceManagementClient, webSiteManagementClient] = await Promise.all([
-    createResourceManagementClient(credentials),
-    createWebSiteManagementClient(credentials),
-  ])
-  const resourceGroupName = createResourceGroupName(config)
-  await createResourceGroup(resourceGroupName, resourceManagementClient)
-  const applicationBuilder = new ApplicationStackBuilder(config)
-  await applicationBuilder.buildOn(logger, resourceManagementClient, webSiteManagementClient, resourceGroupName)
+  logger.info(`Deploying app ${config.appName}`)
+  const zipResource = await terraformSynth(logger, config)
+  const command = await runCommand(process.cwd(), 'cdktf deploy --auto-approve')
+  if (command?.stdout.includes('non-zero exit code')) {
+    return Promise.reject(`Deploy application ${config.appName} failed. Check cdktf logs`)
+  }
+  await uploadFile(logger, config, zipResource)
 }
 
 /**
  * Nuke all the resources used in the Resource Group
  */
 async function nukeApp(_logger: Logger, config: BoosterConfig): Promise<void> {
-  const credentials = await azureCredentials()
-  const resourceManagementClient = await createResourceManagementClient(credentials)
+  const command = await runCommand(process.cwd(), 'cdktf destroy --auto-approve')
+
+  if (command?.stdout.includes('non-zero exit code')) {
+    return Promise.reject(`Destroy application ${config.appName} failed. Check cdktf logs`)
+  }
+}
+
+async function generateSynthFiles(logger: Logger, config: BoosterConfig): Promise<void> {
+  logger.info('Generating cdktf files')
+  const filesToGenerate: Array<[Array<string>, string]> = [[['cdktf.json'], ckdtfTemplate.template]]
+  await Promise.all(filesToGenerate.map(renderToFile(config)))
```

Note that `Promise.all` can fail silently, better use `Promise.allSettledAndFulfilled` (I believe there's a helper function somewhere)

---

Awesome